src/navigation/context.py

```diff
@@ -132,11 +132,12 @@ def other_gate_fid_pos(self) -> Optional[np.ndarray]:
         assert self.ctx.course
         current_waypoint = self.ctx.course.current_waypoint()
         if self.ctx.course.look_for_gate() and current_waypoint is not None:
-            return self.get_fid_pos(current_waypoint.fiducial_id + 1, self.ctx.use_odom)
+            return self.get_fid_pos(self.ctx.use_odom)
         else:
             return None
 
     def current_gate(self, odom_override: bool = True) -> Optional[Gate]:
+        # TODO: Gate is not valid for CIRC. Clean up code, this probably has undefined behavior w/o tag ids
```

Comment:
I think the best thing to do here is just remove this function and basically all parts of the nav code that use it. It should just involve deleting gate.py, partial_gate.py and portions of waypoint.py (when we exit early to go to the gate if we see it) and search.py, and then a search for any other code paths involving gate in the nav node. Or if @Emerson-Ramey want's to go ahead and make these changes in a separate PR I'm fine with that too since it may bloat this PR.

Reply:
Let me know what you too decide

Reply:
I'm thinking we will do all the changes in this one PR since it's just CIRC changes.